roachtest failure: ruby-pg - !!! Installation of ruby 3.1.2 failed!

Since Dec 10, 2022 there have been a number (12 so far) of `ruby-pg` failures. They are categorised as SSH flakes in https://github.com/cockroachdb/cockroach/issues/90695  but upon closer inspection seem to happen because ruby cannot be installed for a reason unrelated to SSH.

It is not consistently failing, but frequently enough to warrant some investigation

```
/usr/local/src/ruby-3.1.2/lib/rubygems/installer.rb:272:in `check_executable_overwrite': "erb" from erb conflicts with /usr/local/bin/erb (Gem::InstallError)
	from ./tool/rbinstall.rb:868:in `check_executable_overwrite'
	from /usr/local/src/ruby-3.1.2/lib/rubygems/installer.rb:513:in `block in generate_bin'
	from /usr/local/src/ruby-3.1.2/lib/rubygems/installer.rb:492:in `each'
	from /usr/local/src/ruby-3.1.2/lib/rubygems/installer.rb:492:in `generate_bin'
	from /usr/local/src/ruby-3.1.2/lib/rubygems/installer.rb:333:in `install'
	from ./tool/rbinstall.rb:899:in `block in install'
	from ./tool/rbinstall.rb:713:in `no_write'
	from ./tool/rbinstall.rb:899:in `install'
	from ./tool/rbinstall.rb:1002:in `block in install_default_gem'
	from ./tool/rbinstall.rb:989:in `each'
	from ./tool/rbinstall.rb:989:in `install_default_gem'
	from ./tool/rbinstall.rb:927:in `block in <main>'
	from ./tool/rbinstall.rb:1127:in `block in <main>'
	from ./tool/rbinstall.rb:1124:in `each'
	from ./tool/rbinstall.rb:1124:in `<main>'
make: *** [uncommon.mk:383: do-install-all] Error 1
!!! Installation of ruby 3.1.2 failed!
```

2 issues:
- ruby is failing installation
- this is categorised as an SSH flake

Jira issue: CRDB-23268

--- a/pkg/cmd/roachtest/tests/ruby_pg.go
+++ b/pkg/cmd/roachtest/tests/ruby_pg.go
@@ -87,6 +87,7 @@ func registerRubyPG(r registry.Registry) {
 			"install ruby 3.1.2",
 			`mkdir -p ruby-install && \
         curl -fsSL https://github.com/postmodern/ruby-install/archive/v0.8.3.tar.gz | tar --strip-components=1 -C ruby-install -xz && \
+        sudo rm -rf /usr/local/bin/* && \
         sudo make -C ruby-install install && \
         sudo ruby-install --system ruby 3.1.2 && \
         sudo gem update --system`,

